hopsworks-api/src/main/java/io/hops/hopsworks/api/kafka/KafkaService.java

```diff
@@ -96,197 +95,143 @@
+  @JWTRequired(acceptedTokens={Audience.API, Audience.JOB}, allowedUserRoles={"HOPS_ADMIN", "HOPS_USER"})
+  public Response getTopic(@Context UriInfo uriInfo, @PathParam("topic") String topicName)
+    throws KafkaException, InterruptedException, ExecutionException {
+    
+    PartitionDetailsDTO dto = topicsBuilder.buildTopicDetails(uriInfo, project, topicName);
+    return Response.ok().entity(dto).build();
   }
-
-  @GET
-  @Path("/topic/defaultValues")
-  @Produces(MediaType.APPLICATION_JSON)
-  @AllowedProjectRoles({AllowedProjectRoles.DATA_OWNER, AllowedProjectRoles.DATA_SCIENTIST})
-  @JWTRequired(acceptedTokens={Audience.API}, allowedUserRoles={"HOPS_ADMIN", "HOPS_USER"})
-  public Response topicDefaultValues() throws KafkaException {
-    TopicDefaultValueDTO values;
-    try {
-      values = kafkaFacade.topicDefaultValues();
-    } catch (InterruptedException | IOException | KeeperException ex) {
-      throw new KafkaException(RESTCodes.KafkaErrorCode.KAFKA_GENERIC_ERROR, Level.SEVERE,
-        "project: " + project.getName(), ex.getMessage(), ex);
-    }
   
-    return noCacheResponse.getNoCacheResponseBuilder(Response.Status.OK).entity(values).build();
-  }
-
-  @GET
-  @Path("/topic/{topic}/share/{projId}")
+  @ApiOperation(value = "Share a Kafka topic with a project.")
+  @PUT
+  @Path("/topics/{topic}/shared/{destProjectId}")
   @Produces(MediaType.APPLICATION_JSON)
   @AllowedProjectRoles({AllowedProjectRoles.DATA_OWNER})
   @JWTRequired(acceptedTokens={Audience.API}, allowedUserRoles={"HOPS_ADMIN", "HOPS_USER"})
-  public Response shareTopic(@PathParam("topic") String topicName, @PathParam("projId") int projectId) throws
-      KafkaException, ProjectException, UserException {
+  public Response shareTopic(@PathParam("topic") String topicName, @PathParam("destProjectId") Integer destProjectId)
+    throws KafkaException, ProjectException, UserException {
+    
+    kafkaController.shareTopicWithProject(project, topicName, destProjectId);
+    
     RESTApiJsonResponse json = new RESTApiJsonResponse();
-    //By default, all members of the project are granted full permissions on the topic
-    Project projectShared = projectFacade.find(projectId);
-    if (projectShared == null) {
-      throw new ProjectException(RESTCodes.ProjectErrorCode.PROJECT_NOT_FOUND, Level.FINE,
-        "Could not find project: " + projectId);
-    }
-    kafkaFacade.shareTopic(project, topicName, projectId);
-
-    AclDTO aclDto = new AclDTO(projectShared.getName(),
-            Settings.KAFKA_ACL_WILDCARD,
-            "allow", Settings.KAFKA_ACL_WILDCARD, Settings.KAFKA_ACL_WILDCARD,
-            Settings.KAFKA_ACL_WILDCARD);
-    kafkaFacade.addAclsToTopic(topicName, project.getId(), aclDto);
     json.setSuccessMessage("The topic has been shared.");
-    return noCacheResponse.getNoCacheResponseBuilder(Response.Status.OK).entity(json).build();
+    return Response.ok().entity(json).build();
```

Comment:
@kai-chi Here we are creating a resource, which is a shared topic. The response code should be 201 and the response should contain the DTO of the created resource and the href to that resource. https://github.com/logicalclocks/hopsworks/blob/v0.10.0/hopsworks-api/src/main/java/io/hops/hopsworks/api/jobs/JobsResource.java#L182